### PR TITLE
Added unit tests in athena-elasticsearch connector

### PR DESCRIPTION
**Pull request by Jithendar12**
*Issue #, if available:*

*Description of changes:*
Added unit tests in athena-elasticsearch connector.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: athena-elasticsearch/src/main/java/com/amazonaws/athena/connectors/elasticsearch/AwsRestHighLevelClientFactory.java [hunk 1]
```diff
@@ -113,7 +113,7 @@ private AwsRestHighLevelClient createClient(String endpoint)
         if (useAwsCredentials) {
             logger.debug("Creating Client using Aws Credentials.");
             return new AwsRestHighLevelClient.Builder(endpoint)
-                    .withCredentials(DefaultCredentialsProvider.create()).build();
+                    .withCredentials(DefaultCredentialsProvider.builder().build()).build();
```
Comment (fal-bharadwaj) [inline athena-elasticsearch/src/main/java/com/amazonaws/athena/connectors/elasticsearch/AwsRestHighLevelClientFactory.java:116]: Why is this change needed in the unit test PR?

File: athena-elasticsearch/src/test/java/com/amazonaws/athena/connectors/elasticsearch/ElasticsearchMetadataHandlerTest.java [hunk 2]
```diff
@@ -549,8 +558,588 @@ public void glueConnectionDomainEndpointWithDomainNameForBackwardCompatibility()
...
+        ElasticsearchMetadataHandler testHandler = createElasticsearchMetadataHandler(config);
+
+        Map<String, String> result = testHandler.resolveDomainMap(config);
+        assertNotNull("Domain map should not be null", result);
+        assertTrue("Should contain movies domain", result.containsKey("movies"));
+        assertEquals("Domain endpoint should match", "https://search-movies.us-east-1.es.amazonaws.com", result.get("movies"));
+    }
+
+    @Test
+    public void doGetTable_withAwsGlueNotNull_retrievesSchemaFromGlue()
```
Comment (fal-bharadwaj) [inline athena-elasticsearch/src/test/java/com/amazonaws/athena/connectors/elasticsearch/ElasticsearchMetadataHandlerTest.java:966]: this test seems like it is testing only mock behavior rather than real use-cases, can we refactor this?

**Review by fal-bharadwaj — CHANGES_REQUESTED** [review]

File: athena-elasticsearch/src/main/java/com/amazonaws/athena/connectors/elasticsearch/AwsRestHighLevelClientFactory.java (hunk 1 above)
Comment (VenkatasivareddyTR) [reply athena-elasticsearch/src/main/java/com/amazonaws/athena/connectors/elasticsearch/AwsRestHighLevelClientFactory.java:116]: Hi @fal-bharadwaj, thanks for the review. This change was needed because DefaultCredentialsProvider.create() is now deprecated in the AWS SDK. The recommended approach is to use the builder pattern — DefaultCredentialsProvider.builder().build() — which ensures forward compatibility with newer SDK versions.

File: athena-elasticsearch/src/test/java/com/amazonaws/athena/connectors/elasticsearch/ElasticsearchMetadataHandlerTest.java (hunk 2 above)
Comment (VenkatasivareddyTR) [reply athena-elasticsearch/src/test/java/com/amazonaws/athena/connectors/elasticsearch/ElasticsearchMetadataHandlerTest.java:966]: Hi @fal-bharadwaj, thanks for the feedback. We’ve refactored the test to mock GlueClient.getTable with a realistic Glue Table (columns, storage descriptor) and to call doGetTable with a real GetTableRequest, so we exercise the actual Glue → Arrow schema conversion in GlueMetadataHandler instead of stubbing super.doGetTable on a spy.

**Review by burhan94 — COMMENTED** [review]
## Code Review

### ✅ Looks Good

Comprehensive tests for AWS request signing, client factory, and caching. Good edge case coverage including null handling and exception scenarios.

### 🟢 Minor suggestion
- Some tests use try-catch+fail() — consider using assertThrows() for cleaner assertions.

**Comment by Jithendar12** [comment]
> ## Code Review
> ### ✅ Looks Good
> 
> Comprehensive tests for AWS request signing, client factory, and caching. Good edge case coverage including null handling and exception scenarios.
> ### 🟢 Minor suggestion
> 
>     * Some tests use try-catch+fail() — consider using assertThrows() for cleaner assertions.

@burhan94, refactored the exception tests to use assertThrows() instead of try-catch+fail().

**Review by burhan94 — COMMENTED** [review]
## Re-Review (April 3)

### ✅ Fixed
- assertThrows() used consistently now — nice cleanup.

### 🟡 Minor (non-blocking)
- A few places still use `anyString()` / `any()` matchers where specific values would be better (AwsRestHighLevelClientFactoryTest, CacheableAwsRestHighLevelClientTest).

Looks good overall. Previous feedback addressed.

**Comment by Jithendar12** [comment]
> ## Re-Review (April 3)
> ### ✅ Fixed
> 
>     * assertThrows() used consistently now — nice cleanup.
> 
> 
> ### 🟡 Minor (non-blocking)
> 
>     * A few places still use `anyString()` / `any()` matchers where specific values would be better (AwsRestHighLevelClientFactoryTest, CacheableAwsRestHighLevelClientTest).
> 
> 
> Looks good overall. Previous feedback addressed.

@burhan94, replaced generic Mockito matchers with specific arguments.

**Review by aimethed — APPROVED** [review]

**Review by hareeshkumar-t — APPROVED** [review]

**Review by fal-bharadwaj — APPROVED** [review]
